tls/s2n_client_hello.c

```diff
@@ -971,3 +972,33 @@ int s2n_client_hello_has_extension(struct s2n_client_hello *ch, uint16_t extensi
     }
     return S2N_SUCCESS;
 }
+
+int s2n_client_hello_get_supported_groups(struct s2n_client_hello *ch, uint16_t *groups,
+        uint16_t groups_count_max, uint16_t *groups_count_out)
+{
+    POSIX_ENSURE_REF(groups_count_out);
+    *groups_count_out = 0;
+    POSIX_ENSURE_REF(ch);
+    POSIX_ENSURE_REF(groups);
+
+    s2n_parsed_extension *supported_groups_extension = NULL;
+    POSIX_GUARD(s2n_client_hello_get_parsed_extension(S2N_EXTENSION_SUPPORTED_GROUPS, &ch->extensions, &supported_groups_extension));
```

Comment:
Added, thanks. It's not currently possible for this to happen since the s2n_parsed_extension is stack allocated in the s2n_parsed_extensions_list. If the extension wasn't received the blob data will be null, and is already checked in s2n_client_hello_get_parsed_extension. So I kept the safety error.